### PR TITLE
timeline creation: reflect failures due to ancestor LSN issues in status code

### PR DESCRIPTION
**Pull request by problame**
Before, it was a `500` and control plane would retry, wereas it actually
should have stopped retrying.

(Stacked on top of https://github.com/neondatabase/neon/pull/4597 )

fixes https://github.com/neondatabase/neon/issues/4595
part of https://github.com/neondatabase/cloud/issues/5626

### CODE REVIEW

File: pageserver/src/tenant.rs [hunk 1]
```diff
@@ -3824,6 +3827,9 @@ mod tests {
         {
             Ok(_) => panic!("branching should have failed"),
             Err(err) => {
+                let CreateTimelineError::AncestorLsn(err) = err else {
```
[the remark this answers is not in the document]
Comment (problame) [reply pageserver/src/tenant.rs:3830]: Not sure I follow. I need the destructuring, how will `Eq` help?

File: pageserver/src/tenant.rs [hunk 2]
```diff
@@ -2782,16 +2784,17 @@ impl Tenant {
...
+            ))
+            .map_err(CreateTimelineError::AncestorLsn)?;
 
         // and then the planned GC cutoff
         {
             let gc_info = src_timeline.gc_info.read().unwrap();
             let cutoff = min(gc_info.pitr_cutoff, gc_info.horizon_cutoff);
             if start_lsn < cutoff {
-                bail!(format!(
+                return Err(CreateTimelineError::AncestorLsn(anyhow::format_err!(
```
[the remark this answers is not in the document]
Comment (problame) [reply pageserver/src/tenant.rs:2795]: Yeah, I think they're identical / `anyhow::anyhow` is preferred. Will fix.

File: pageserver/src/tenant.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (LizardWizzard) [reply pageserver/src/tenant.rs:3831]: I suggest to use `assert_eq!` to check that this is the right error. assert_eq needs that items you pass to it implement Eq

[the remark this answers is not in the document]
Comment (problame) [reply pageserver/src/tenant.rs:3831]: Yeah but is anyhow `Eq`? Don't get how it would help?

[the remark this answers is not in the document]
Comment (LizardWizzard) [reply pageserver/src/tenant.rs:3831]: You want to check that you got specific variant of `CreateTimelineError`. `CreateTimelineError` is not `Eq`

[the remark this answers is not in the document]
Comment (problame) [reply pageserver/src/tenant.rs:3831]: Resolved offline.

**Pull request merged**